engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -751,6 +771,20 @@ void paladin_t::trigger_holy_shield( action_state_t* s )
   active.holy_shield_damage -> schedule_execute();
 }
 
+void paladin_t::trigger_t28_4p_pp( action_state_t* s )
+{
+  // escape if we don't have t28 4p
+  // if ( !talents.holy_shield->ok() )
+  //  return;
+
+  // sanity check - no friendly-fire
+  if ( !s->action->player->is_enemy() )
+    return;
+
```

Comment:
You can add the following here, makes debugging easier if your trigger and proc action creation conditions ever get out of sync:
`assert( active.t28_4p_pp != nullptr );`